infra/ci/build.py

```diff
@@ -168,6 +189,42 @@ def main():
 
   if failed_projects:
     print('Failed projects:', ' '.join(failed_projects))
+    return BuildModifiedProjectsResult.BUILD_FAIL
+
+  return BuildModifiedProjectsResult.BUILD_SUCCESS
+
+
+def should_build_canary_project():
+  """Returns True if we should build the canary project."""
+  git_output = get_changed_files()
+  infra_code_regex = '.*infra/.*\n'
+  return re.search(infra_code_regex, git_output) is not None
+
+
+def build_canary_project():
+  """Builds a specific project when infra/ is changed to verify that infra/
+  changes don't break things. Returns False if build was attempted but
+  failed."""
+  if not should_build_canary_project():
+    return True
+
+  try:
+    build_project('skcms')
+  except subprocess.CalledProcessError:
+    return False
+
+  return True
+
+
+def main():
+  """Build modified projects or canary project."""
+  result = build_modified_projects()
+  if result == BuildModifiedProjectsResult.BUILD_FAIL:
+    return 1
+
+  if (result == BuildModifiedProjectsResult.NONE_BUILT and
+      not build_canary_project()):
+    # It's unnecessary to build the canry
```

Comment:
typo: canry
is this comment correct, i think this branch means canary project failed no ?